server_test.go

```diff
@@ -50,17 +52,26 @@ func AnotherHelloServer(w ResponseWriter, req *Msg) {
 }
 
 func RunLocalUDPServer(laddr string) (*Server, string, error) {
-	server, l, _, err := RunLocalUDPServerWithFinChan(laddr)
+	server, l, _, err := RunLocalUDPServerWithFinChan(laddr, 0)
 
 	return server, l, err
 }
 
-func RunLocalUDPServerWithFinChan(laddr string) (*Server, string, chan error, error) {
+func RunLocalUDPServerWithFinChan(laddr string, rdtimeout time.Duration) (*Server, string, chan error, error) {
 	pc, err := net.ListenPacket("udp", laddr)
 	if err != nil {
 		return nil, "", nil, err
 	}
-	server := &Server{PacketConn: pc, ReadTimeout: time.Hour, WriteTimeout: time.Hour}
+
+	// tune the timeouts. If shutdown is lower than rdtimeout and there is no queries,
+	// then ActivateAndServe will end on an error because connection will be close before timeout of ReadUDP.
+
+	shtimeout := rdtimeout + (1 * time.Second)  // to avoid error return by Server, we need that ShutdownTimeout is > ReadTimeout
+	if rdtimeout == 0 {
+		rdtimeout = 1 * time.Hour
```

Comment:
hour?

Reply:
I will change with an option rdtimeout in RunLocalUDPServerWithFinChan as suggested above.
All values were formerly in hours ...